plugins/modules/device_credential_intent.py

```diff
@@ -0,0 +1,2323 @@
+
+        return self
+
+    def delete_device_credential(self, config):
+        """
+        Delete Global Device Credential in Cisco DNA Center based on the playbook details.
+        Check the return value of the API with check_return_status().
+
+        Parameters:
+            None
+
+        Returns:
+            None
+        """
+
+        result_global_credential = self.result.get("response")[0].get("globalCredential")
+        have_values = self.have.get("globalCredential")
+        final_response = {}
+        self.log(str(have_values))
+        for item in have_values:
+            config_itr = 0
+            final_response.update({item: []})
+            for value in have_values.get(item):
+                if value is None:
+                    self.msg = str(config.get("GlobalCredentialDetails")
+                                   .get("item")[config_itr]) + "is not found"
+                    self.status = "failed"
+                    return self
+                _id = have_values.get(item)[config_itr].get("id")
+                response = self.dnac._exec(
+                    family="discovery",
+                    function="delete_global_credential_v2",
+                    params={"id": _id},
+                )
+                self.log(str(response))
+                if not response:
+                    self.msg = "response is empty"
+                    self.status = "exited"
+                    return self
+
+                if not isinstance(response, dict):
+                    self.msg = "response is not a dictionary"
+                    self.status = "exited"
+                    return self
+
+                task_id = response.get("response").get("taskId")
+                while True:
+                    task_details = self.get_task_details(task_id)
+                    if task_details.get("isError") is True:
+                        self.msg = str(task_details.get("progress"))
+                        self.status = "failed"
+                        return self
+                    elif task_details.get("isError") is False:
+                        self.result['changed'] = True
+                        break
+                final_response.get(item).append(_id)
+                config_itr = config_itr + 1
+
+        self.log("Delete Device Credential API input - " + str(final_response))
+        self.log("Global Device Credential Deleted Successfully")
+        result_global_credential.update({
+            "Deletion": {
+                "response": final_response,
+                "msg": "Global Device Credentials Deleted Successfully"
+            }
+        })
+        self.msg = "Global Device Credentials Updated Successfully"
+        self.status = "success"
+        return self
+
+    def get_diff_deleted(self, config):
+        """
+        Delete Global Device Credential in Cisco DNA Center based on the playbook details.
+
+        Parameters:
+            None
```

Comment:
config (dict) - Playbook details containing Global Device Credential information.
and self